migration_sql.go

```diff
@@ -154,6 +154,12 @@ func runSQLMigration(db *sql.DB, scriptFile string, v int64, direction bool) err
 			log.Fatal(err)
 		}
 
+		err = GetDialect().lockTableQuery(tx)
```

Comment:
This will error if the migration runs outside of a transaction for postgres, maybe we should check that we are running a migration with transaction before attempting to acquire lock.

Reply:
Are we not guaranteed to be in a transaction based from line 152, assumed we do no error out on line 154?

Reply:
@KillerX Ah, you're correct. I missed the `if useTx {`

Reply:
what if we're outside of Tx, do we still lock the table?

Reply:
asking because we support `-- +goose NO TRANSACTION`

Reply:
If we're outside of tx, the table will not be locked. Postgres doesn't support locks outside of transactions.

Reply:
I see